### PR TITLE
Add RPC client library

### PR DESCRIPTION
**Pull request by GraDKh**
# Issue ticket

Issue ticket link: <>

### Code conventions

- [x] I have performed a self-review of my code
- [x] Every new function is documented
- [x] Object names are auto explicative

### Security

- [x] The PR does not break APIs backward compatibility
- [x] The PR does not break the stable storage backward compatibility

### Testing

- [x] Every function is properly unit tested
- [x] I have added integration tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] IC endpoints are always tested through the `canister_call!` macro

### CODE REVIEW

**Review by F3kilo — APPROVED** [review]

File: src/rpc-client/src/request.rs [hunk 1]
```diff
@@ -0,0 +1,120 @@
+use anyhow::Context;
+use itertools::Itertools;
+use jsonrpc_core::{Call, Id, MethodCall, Output, Params, Request, Response, Version};
+use serde::Deserialize;
+
+const MAX_BATCH_REQUESTS: usize = 5; // Max batch size is 5 in EVM
```
Comment (ufoscout) [inline src/rpc-client/src/request.rs:6]: it is 50 in evm

File: Cargo.toml [hunk 2]
```diff
@@ -7,11 +7,12 @@ members = [
...
     "src/evm-canister-client",
     "src/iceth-client",
-    "src/minter-did",
-    "src/minter-client",
     "src/iceth-client/test-canister",
+    "src/minter-client",
+    "src/minter-did",
     "src/register-evm-agent/src/register_evm_agent",
     "src/register-evm-agent/src/simple_canister",
+    "src/rpc-client",
```
Comment (ufoscout) [inline Cargo.toml:15]: should be `ethereum_json_rpc_api_client`

**Review by ufoscout — CHANGES_REQUESTED** [review]

File: src/rpc-client/src/request.rs (hunk 1 above)
Comment (GraDKh) [reply src/rpc-client/src/request.rs]: Updated

File: Cargo.toml (hunk 2 above)
Comment (GraDKh) [reply Cargo.toml]: If you don't mind, I've changed it to `ethereum-json-rpc-client` to make it a bit shorter

Comment (ufoscout) [reply Cargo.toml]: perfect

**Review by ufoscout — APPROVED** [review]

**Pull request merged**